.pre-commit-hooks.yaml

```
@@ -0,0 +1,24 @@
+- id: poetry-check
+  name: poetry-check
+  description: run poetry check to validate config
+  entry: poetry check
+  language: python
+  language_version: python3
+  pass_filenames: false
```

I think you need
```suggestion
  pass_filenames: false
  require_serial: true
  files: ^*/(pyproject\.toml|poetry.lock)$
```

on all of these

---

Why do you think I need this?

Line 8: By default this setting is `false` but why switching it? The hook calls `poetry check` which is one process in this context anyway. So I see no improvement here.

Line 9: `poetry check` does not take arguments like file names. Therefore this results in an `Too many arguments.` error. Same reason why `pass_filenames: false` is set.

I also run the exact same config from this PR in my [temp. mirror repo](https://github.com/Cielquan/mirrors-poetry) and it just works fine.

---

without this I get:

```
poetry-check.........................................(no files to check)Skipped
poetry-lock..........................................(no files to check)Skipped
poetry-export........................................(no files to check)Skipped
```

---

For me its working just fine.

relevant `.pre-commit-config.yaml` part:
```yaml
repos:
  - repo: https://github.com/Cielquan/mirrors-poetry
    rev: 87bf4eb48cd53dc409165358a84b1687b03efc9c  # frozen: 1.2.0a1
    hooks:
      - id: poetry-check
      - id: poetry-lock
      - id: poetry-export
```
Output:
```console
poetry-check.............................................................Passed
poetry-lock..............................................................Failed
- hook id: poetry-lock
- files were modified by this hook

Updating dependencies
Resolving dependencies... (15.7s)

Writing lock file

poetry-export............................................................Passed
```

I really do not understand your output, because if you use "my config", `pre-commit` does not send filenames to the hook and therefore does not require any and therefore should never show `(no files to check)Skipped`. When I delete my `pyproject.toml` file I get the error below for all 3 hooks. And this error comes from the hook and not `pre-commit`.
```console
poetry-check.............................................................Failed
- hook id: poetry-check
- exit code: 1

RuntimeError

  Poetry could not find a pyproject.toml file in /home/krys/Projects/.testing/python_test-cielquan or its parents

  at ~/.cache/pre-commit/repoqhauindm/py_env-python3/lib/python3.8/site-packages/poetry/core/factory.py:395 in locate
      391│             if poetry_file.exists():
      392│                 return poetry_file
      393│ 
      394│         else:
    → 395│             raise RuntimeError(
      396│                 "Poetry could not find a pyproject.toml file in {} or its parents".format(
      397│                     cwd
      398│                 )
      399│             )

```

---

Are you running with `pre-commit run --all-files`?

---

Yes I did for the answer above but after reading you question I ran and got
```console
$ pre-commit install
$ touch tesfile
$ git add testfile
$ git commit -m "add testfile"
poetry-check.............................................................Passed
poetry-lock..............................................................Passed
poetry-export............................................................Passed
...
```

---

yes, it always runs if any files change, making pre-commit *much* slower

---

@graingert is correct here (though `require_serial` is implied by `pass_filenames: false`) -- it should specify `files:` so it doesn't run all the time

---

Ah ok .. I get the reasoning now. Thanks.
I will add `files: '^.*/pyproject\.toml$'`  for `check` and `files: '^.*/poetry\.lock$'` for `export` hook.